### PR TITLE
Issue #2109: CLI should print a file name where exception is happen

### PR DESCRIPTION
**Pull request by romani**

### CODE REVIEW

**Comment by mkordas** [comment]
@romani 
This time Appveyor complains due to platform-specific newline characters.

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java [hunk 1]
```diff
@@ -252,28 +252,35 @@ public int process(List<File> files) throws CheckstyleException {
...
+                fireErrors(fileName, fileMessages);
+                fireFileFinished(fileName);
             }
-            catch (final IOException ioe) {
-                LOG.debug("IOException occurred.", ioe);
-                fileMessages.add(new LocalizedMessage(0,
-                        Definitions.CHECKSTYLE_BUNDLE, "general.exception",
-                        new String[] {ioe.getMessage()}, null, getClass(),
-                        null));
+            catch (Exception ex) {
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: No need to catch checked exceptions here, `RuntimeException` is enough.

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java [hunk 2]
```diff
@@ -252,28 +252,35 @@ public int process(List<File> files) throws CheckstyleException {
...
+                // We need to catch all exception to put a reason failure(file name) in exception
+                throw new CheckstyleException("Exception happens during processing of "
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: I would propose `Exception was thrown while processing `

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java (hunk 1 above)
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: yes, I thought about this, but after extra time, It is better to catch Exception there as one day Checked exception will happen (even after some refactoring) in this block and developer will update a method signature but he will miss this catch for sure.

I thought to catch Throwable there as 100% guaranty that on any problem we will report file name ( we could catch OOM Exception on some files eventually), but smth telling me to stay with Exception is ok, and all Errors are problems that most likely do not need filename.

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java (hunk 2 above)
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: Right, that is better wording, fixed.

**Comment by romani** [comment]
@mkordas , I can not do "git push --force" in checkcstyle as it is forbidden for all branches for security reason. The only workaround is to drop this PR and create new PR. I do not want to do this, I will squash all in oner commit on a moment of merge to master. 

Please review this PR, I am going to merge it in 8 hours and start release 6.12 process.

**Comment by romani** [comment]
merged as FF

**Pull request closed without merging**